apps/backend/src/users/users.service.ts

```diff
@@ -21,4 +25,55 @@ export class UsersService {
   remove(id: number) {
     return `This action removes a #${id} user`;
   }
+
+  async addFavoriteDrink(userId: string, favoriteDrinkId: string): Promise<User> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId: userId },
+      include: { favouriteDrinks: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    if (user.favouriteDrinks.length >= 3) {
+      throw new BadRequestException('A user can have only 3 favorite drinks.');
+    }
+
+    if (user.favouriteDrinks.some((drink) => drink.id === favoriteDrinkId)) {
+      throw new BadRequestException('The drink is already in the user list of favorite drinks.');
+    }
+
+    const updatedUser = await this.prisma.user.update({
+      where: { authSchId: userId },
+      data: { favouriteDrinks: { connect: { id: favoriteDrinkId } } },
+      include: { favouriteDrinks: true },
+    });
+
+    return updatedUser;
+  }
+
+  async removeFavoriteDrink(userId: string, favoriteDrinkId: string): Promise<User> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId: userId },
+      include: { favouriteDrinks: true },
+    });
```

Comment:
just a suggestion: you can filter the records to include inside the prisma queries. So you could include only the drinks where the id === favouriteDrinkId, and then you'd only have to check if the array is empty or not.